packages/batch-submitter/src/batch-submitter/state-batch-submitter.ts

```diff
@@ -159,48 +155,38 @@ export class StateBatchSubmitter extends BatchSubmitter {
     endBlock: number
   ): Promise<TransactionReceipt> {
     const batch = await this._generateStateCommitmentBatch(startBlock, endBlock)
-    const tx = this.chainContract.interface.encodeFunctionData(
+    const calldata = this.chainContract.interface.encodeFunctionData(
       'appendStateBatch',
       [batch, startBlock]
     )
-    const batchSizeInBytes = remove0x(tx).length / 2
+    const batchSizeInBytes = remove0x(calldata).length / 2
```

Comment:
I think there is utility in `ethers` called `hexDataLength` we could use here.

Reply:
Ideally everything was `Buffer`s instead of `string`s or even uint8 arrays so that we didn't have to deal with constant prefix slicing